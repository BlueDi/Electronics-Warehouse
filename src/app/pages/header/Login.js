import React, { Component } from 'react';
import { Menu, Modal } from 'semantic-ui-react';
import LoginForm from './LoginForm';
import SignupForm from './SignupForm';

class LoginModal extends Component {
  render() {
    return (
      <Modal size="tiny" trigger={<Menu.Item>Login</Menu.Item>}>
        <Modal.Content>
          <LoginForm />
        </Modal.Content>
      </Modal>
    );
  }
}

class SignupModal extends Component {
  render() {
    return (
      <Modal size="small" trigger={<Menu.Item>Signup</Menu.Item>}>
        <Modal.Content>
          <SignupForm />
        </Modal.Content>
      </Modal>
    );
  }
}

class Login extends Component {
  render() {
    return (
      <Menu.Menu position="right">
        <SignupModal />
        <LoginModal />
      </Menu.Menu>
    );
  }
}

export { LoginModal, SignupModal };
export default Login;
