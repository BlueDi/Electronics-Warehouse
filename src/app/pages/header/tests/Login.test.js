import React from 'react';
import renderer from 'react-test-renderer';
import Login, { LoginModal, SignupModal } from '@pages/header/Login';

describe('<Login />', () => {
  test('Renders properly', () => {
    const component = renderer.create(<Login />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<LoginModal />', () => {
  test('Renders properly', () => {
    const component = renderer.create(<LoginModal />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<SignupModal />', () => {
  test('Renders properly', () => {
    const component = renderer.create(<SignupModal />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
