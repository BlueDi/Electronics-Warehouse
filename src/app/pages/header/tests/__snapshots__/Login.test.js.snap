// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Renders properly 1`] = `
<div
  className="right menu"
>
  <div
    className="item"
    onClick={[Function]}
  >
    Signup
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    Login
  </div>
</div>
`;

exports[`<LoginModal /> Renders properly 1`] = `
<div
  className="item"
  onClick={[Function]}
>
  Login
</div>
`;

exports[`<SignupModal /> Renders properly 1`] = `
<div
  className="item"
  onClick={[Function]}
>
  Signup
</div>
`;
